{'bins': 16, 'cliffs': 0.147, 'file': 'data/auto93.csv', 'Far': 0.95, 'go': 'all', 'help': False, 'Halves': 512, 'min': 0.5, 'Max': 512, 'p': 2, 'rest': 4, 'Reuse': True, 'seed': 937162211, 'debug': False}
✅ pass:	 the
✅ pass:	 rand
{}
✅ pass:	 some
Error in example 'nums': module 'lists' has no attribute 'add'
Traceback (most recent call last):
  File "/home/runner/work/engr-csc591-021-spring2023/engr-csc591-021-spring2023/src/HW5/run.py", line 35, in make_attempt
    passed = funs[what]()
  File "/home/runner/work/engr-csc591-021-spring2023/engr-csc591-021-spring2023/src/HW5/examples.py", line 65, in eg_function_2
    lists.add(num1, numerics.rand())
AttributeError: module 'lists' has no attribute 'add'

❌ fail:	 nums
Error in example 'syms': module 'lists' has no attribute 'adds'
Traceback (most recent call last):
  File "/home/runner/work/engr-csc591-021-spring2023/engr-csc591-021-spring2023/src/HW5/run.py", line 35, in make_attempt
    passed = funs[what]()
  File "/home/runner/work/engr-csc591-021-spring2023/engr-csc591-021-spring2023/src/HW5/examples.py", line 73, in eg_function_3
    sym=lists.adds(Sym.Sym(), {"a","a","a","a","b","b","c"})
AttributeError: module 'lists' has no attribute 'adds'

❌ fail:	 syms
Error in example 'csv': local variable 'n' referenced before assignment
Traceback (most recent call last):
  File "/home/runner/work/engr-csc591-021-spring2023/engr-csc591-021-spring2023/src/HW5/run.py", line 35, in make_attempt
    passed = funs[what]()
  File "/home/runner/work/engr-csc591-021-spring2023/engr-csc591-021-spring2023/src/HW5/examples.py", line 82, in eg_function_4
    strings.csv(d.the["file"], fun)
  File "/home/runner/work/engr-csc591-021-spring2023/engr-csc591-021-spring2023/src/HW5/strings.py", line 102, in csv
    fun(t)
  File "/home/runner/work/engr-csc591-021-spring2023/engr-csc591-021-spring2023/src/HW5/examples.py", line 80, in fun
    n += len(t)
UnboundLocalError: local variable 'n' referenced before assignment

❌ fail:	 csv
Error in example 'data': 'Num' object has no attribute 'get'
Traceback (most recent call last):
  File "/home/runner/work/engr-csc591-021-spring2023/engr-csc591-021-spring2023/src/HW5/run.py", line 35, in make_attempt
    passed = funs[what]()
  File "/home/runner/work/engr-csc591-021-spring2023/engr-csc591-021-spring2023/src/HW5/examples.py", line 88, in eg_function_5
    print(col.lo,col.hi, query.mid(col), query.div(col))
  File "/home/runner/work/engr-csc591-021-spring2023/engr-csc591-021-spring2023/src/HW5/query.py", line 11, in mid
    if col.get("isSym"):
AttributeError: 'Num' object has no attribute 'get'

❌ fail:	 data
{'Lbs-': 2970.42, 'Acc+': 15.57, 'Mpg+': 23.84}
{'Lbs-': 2970.42, 'Acc+': 15.57, 'Mpg+': 23.84}
✅ pass:	 clone
Error in example 'cliffs': name 'cliffsDelta' is not defined
Traceback (most recent call last):
  File "/home/runner/work/engr-csc591-021-spring2023/engr-csc591-021-spring2023/src/HW5/run.py", line 35, in make_attempt
    passed = funs[what]()
  File "/home/runner/work/engr-csc591-021-spring2023/engr-csc591-021-spring2023/src/HW5/examples.py", line 98, in eg_function_7
    assert(False == cliffsDelta( {8,7,6,2,5,8,7,3},{8,7,6,2,5,8,7,3}),"1")
NameError: name 'cliffsDelta' is not defined

❌ fail:	 cliffs
Error in example 'dist': 'int' object has no attribute 'cells'
Traceback (most recent call last):
  File "/home/runner/work/engr-csc591-021-spring2023/engr-csc591-021-spring2023/src/HW5/run.py", line 35, in make_attempt
    passed = funs[what]()
  File "/home/runner/work/engr-csc591-021-spring2023/engr-csc591-021-spring2023/src/HW5/examples.py", line 119, in eg_function_8
    num.add(data.dist(row, data.rows[1]))
  File "/home/runner/work/engr-csc591-021-spring2023/engr-csc591-021-spring2023/src/HW5/Data.py", line 98, in dist
    d = d + (col.dist(row1.cells[col.at], row2.cells[col.at])**g.the.get("p"))
AttributeError: 'int' object has no attribute 'cells'

❌ fail:	 dist
Error in example 'half': too many values to unpack (expected 5)
Traceback (most recent call last):
  File "/home/runner/work/engr-csc591-021-spring2023/engr-csc591-021-spring2023/src/HW5/run.py", line 35, in make_attempt
    passed = funs[what]()
  File "/home/runner/work/engr-csc591-021-spring2023/engr-csc591-021-spring2023/src/HW5/examples.py", line 127, in eg_function_9
    left,right,A,B,c = data.half()
ValueError: too many values to unpack (expected 5)

❌ fail:	 half
Error in example 'tree': name 'showTree' is not defined
Traceback (most recent call last):
  File "/home/runner/work/engr-csc591-021-spring2023/engr-csc591-021-spring2023/src/HW5/run.py", line 35, in make_attempt
    passed = funs[what]()
  File "/home/runner/work/engr-csc591-021-spring2023/engr-csc591-021-spring2023/src/HW5/examples.py", line 134, in eg_function_10
    showTree(tree(Data.Data(d.the.file)))
NameError: name 'showTree' is not defined

❌ fail:	 tree
Error in example 'sway': 'Data' object has no attribute 'sway'
Traceback (most recent call last):
  File "/home/runner/work/engr-csc591-021-spring2023/engr-csc591-021-spring2023/src/HW5/run.py", line 35, in make_attempt
    passed = funs[what]()
  File "/home/runner/work/engr-csc591-021-spring2023/engr-csc591-021-spring2023/src/HW5/examples.py", line 138, in eg_function_11
    best,rest = data.sway()
AttributeError: 'Data' object has no attribute 'sway'

❌ fail:	 sway
Error in example 'bins': 'Data' object has no attribute 'sway'
Traceback (most recent call last):
  File "/home/runner/work/engr-csc591-021-spring2023/engr-csc591-021-spring2023/src/HW5/run.py", line 35, in make_attempt
    passed = funs[what]()
  File "/home/runner/work/engr-csc591-021-spring2023/engr-csc591-021-spring2023/src/HW5/examples.py", line 150, in eg_function_12
    best,rest = data.sway()
AttributeError: 'Data' object has no attribute 'sway'

❌ fail:	 bins
#W ?k <class 'str'>
#W ?v <class 'dict'>
#W ?settings <class 'function'>
#W ?cli <class 'function'>
#W ?make_attempt <class 'function'>
#W ?main <class 'function'>
#W ?Seed <class 'int'>
✅ Number of tests passing -> 4
❌ Number of tests failing -> 10
